pandas/tests/indexes/multi/test_drop.py

```diff
@@ -139,3 +139,22 @@ def test_drop_not_lexsorted():
     tm.assert_index_equal(lexsorted_mi, not_lexsorted_mi)
     with tm.assert_produces_warning(PerformanceWarning):
         tm.assert_index_equal(lexsorted_mi.drop("a"), not_lexsorted_mi.drop("a"))
+
+
+def test_drop_raise_exception_if_labels_not_in_level():
+    # GH 8594
+    mi = MultiIndex.from_arrays([[1, 2, 3], [4, 5, 6]], names=["a", "b"])
+    s = pd.Series([10, 20, 30], index=mi)
+    df = pd.DataFrame([10, 20, 30], index=mi)
+
+    msg = r"labels \[4\] not found in level"
+    with pytest.raises(KeyError, match=msg):
+        s.drop(4, level="a")
+    with pytest.raises(KeyError, match=msg):
+        df.drop(4, level="a")
+
+    msg = r"labels \[7\] not found in level"
+    with pytest.raises(KeyError, match=msg):
+        s.drop(7, level="b")
+    with pytest.raises(KeyError, match=msg):
+        df.drop(7, level="b")
```

Comment:
We can probably parameterize this (via `pytest` idiom).  Not necessary to do now though.